### PR TITLE
Added benchmarking for rosetta-cli

### PR DESCRIPTION
**Pull request by raghavapamula**
Added setup for benchmarking. Added benchmarking for Sync() method from rosetta-sdk. Benchmarks can be run with the check:benchmark command.

### Motivation
Optimizing rosetta-cli and rosetta-sdk

### CODE REVIEW

**Review by nasmithan — APPROVED** [review]

**Comment by cb-heimdall** [comment]
Review Error for nasmithan @ 2022-03-24 02:14:43 UTC
User must have write permissions to review

File: pkg/tester/data.go [hunk 1]
```diff
@@ -220,7 +220,7 @@ func InitializeData(
 	}
 
 	networkOptions, fetchErr := fetcher.NetworkOptionsRetry(ctx, network, nil)
-	if err != nil {
+	if fetchErr != nil {
 		log.Fatalf("%s: unable to get network options", fetchErr.Err.Error())
```
Comment (racbc) [inline pkg/tester/data.go:224]: Same question as above.

File: pkg/tester/data_bench.go [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+		config.Data.LogBalanceChanges,
+		config.Data.LogReconciliations,
+		logger.Data,
+		config.Network,
+	)
+
+	localStore, err := database.NewBadgerDatabase(ctx, dataPath)
+
+	if err != nil {
+		logger.Error("%s: cannot initialize database")
```
Comment (racbc) [inline pkg/tester/data_bench.go:58]: Same question as above.

File: cmd/benchmark.go [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+	"context"
+	"fmt"
+	t "github.com/coinbase/rosetta-cli/pkg/tester"
+	"github.com/spf13/cobra"
+)
+
+var (
+	checkBenchmarkCmd = &cobra.Command{
+		Use:   "check:benchmark",
+		Short: "Benchmark performance of time critical methods of rosetta-cli and rosetta-sdk",
```
Comment (racbc) [inline cmd/benchmark.go:13]:
```suggestion
		Short: "Benchmark performance of time-critical methods of rosetta-cli and rosetta-sdk",
```

File: cmd/benchmark.go [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+		Long: `This command can be used to benchmark the performance of time critical methods for rosetta-cli and rosetta-sdk.
+This is useful for optimizing the performance of commands like check:data and check:construction, which can take a long time to complete.
+The performance improvement can be objectively compared.`,
```
Comment (racbc) [inline cmd/benchmark.go:16]: - "Time-critical" needs a dash.
- Is the correct term `rosetta-sdk` or `rosetta-sdk-go`?

```suggestion
		Long: `This command can be used to benchmark the performance of time-critical methods for rosetta-cli and rosetta-sdk.
This is useful for optimizing the performance of commands like check:data and check:construction, which can take a long time to complete.
The performance improvement can be objectively compared.`,
```

"The performance improvement can be objectively compared." - What does this mean? How would a user compare the improvement? Or is that something that automatically happens once a user runs check:benchmark?

File: pkg/tester/construction.go [hunk 5]
```diff
@@ -103,7 +103,7 @@ func InitializeConstruction(
 	}
 
 	networkOptions, fetchErr := onlineFetcher.NetworkOptionsRetry(ctx, network, nil)
-	if err != nil {
+	if fetchErr != nil {
 		log.Fatalf("%s: unable to get network options", fetchErr.Err.Error())
```
[the remark this answers is not in the document]
Comment (raghavapamula) [reply pkg/tester/construction.go:107]: i think the rosetta server should return the network options. If this gives an error, probably there is something wrong with the rosetta-server/implementation

File: pkg/tester/data_bench.go (hunk 2 above)
Comment (raghavapamula) [reply pkg/tester/data_bench.go:58]: This error would be a problem with BadgerDB, which is an external golang module

**Pull request closed without merging**